crates/data-transformer/src/calldata.rs

```diff
@@ -0,0 +1,47 @@
+use super::transformer::transform;
+use serde::{Deserialize, Serialize};
+use starknet::core::types::{ContractClass, Felt};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Calldata {
```

Comment:
inconsistent with design doc

Reply:
Please explain why or change this

Reply:
We cannot implement this exactly as stated in the doc. The problem lies in the tokenization and parsing of a sequence of quoted expressions. We need some dedicated delimiter (other than space) separating expressions.

Reply:
Parsing quoted expressions separated by whitespace is problematic. `ByteArray`s written in quotes would be tough to parse, especially if they contain escaped quotes and whitespace. Parsing them properly is either impossible with regexes or require exceptional tricks. May even require implementing a custom automaton. Modyfing the design by adding comma-separation allows us to delegate whole expression processing logic to an already existing Cairo parser.